Added note for ece-tools patch process fix (#6004)
magento/devdocs: guides/v2.2/cloud/release-notes/cloud-tools.md

   -  {:.new}Updated the `composer.json` file for the `{{ site.data.var.ct }}` package to add a dependency for the `magento/magento-cloud-patches` v1.0.0 package.

>    -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to fail when applying security-only patches for versions 2.3.2 and later. This issue was introduced by the new versioning scheme adopted for security-only patches in {{ site.data.var.ee }} v2.3.3. See [Security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches).

Reviewer: Maybe specify "Magento version 2.3.2 and later"

>    -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to fail when applying security-only patches for versions 2.3.2 and later releases. This issue was introduced by the new versioning scheme adopted for security-only patches in {{ site.data.var.ee }} v2.3.3. See [Security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches).

Author: ```suggestion
   -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to fail when applying security-only patches for versions 2.3.2 and later. This issue was introduced by the new versioning scheme adopted for security-only patches in {{ site.data.var.ee }} v2.3.3. See [Security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches).
```

>    -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to fail when applying security-only patches for versions 2.3.2 and later releases. This issue was introduced by the new versioning scheme adopted for security-only patches in {{ site.data.var.ee }} v2.3.3. See [Security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches).

Reviewer 2: Not `fail` but `skip patches, which were expected to apply, causing no patches to be applied at all`. This can be better re-worded, it's just the issue itself.

Also I would change `...security-only patches for versions 2.3.2 and later releases.` to `...2.3.2-p2 release and any future security-only releases.`

Let's call `security-only patches` as `security-only releases` to avoid confusion with actual patches (hotfixes), which we fixed. Also new scheme was adopted starting 2.3.2 with 2.3.2-p2 release.

>    -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to fail when applying security-only patches for versions 2.3.2, 2.3.3, and later releases. This issue was introduced by the new versioning scheme adopted for security-only patches in {{ site.data.var.ee }} v2.3.3. See [Security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches).

Author: ```suggestion
   -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to fail when applying security-only patches for versions 2.3.2 and later releases. This issue was introduced by the new versioning scheme adopted for security-only patches in {{ site.data.var.ee }} v2.3.3. See [Security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches).
```

>    -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to break when applying patch sets that include security-only patches for Magento version 2.3.2 and later . This issue was introduced by the new versioning scheme adopted for [security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches) starting with the {{ site.data.var.ee }} version 2.3.2-p2.

Reviewer 2: `when applying patch sets that include security-only patches for Magento version 2.3.2 and later ` -> `when applying patch sets on top of security-only releases, starting with Magento 2.3.2-p2 and later`

>    -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to break when applying patch sets that include security-only patches for Magento version 2.3.2 and later . This issue was introduced by the new versioning scheme adopted for [security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches) starting with the {{ site.data.var.ee }} version 2.3.2-p2.

Author: ```suggestion
   -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to break when applying patch sets on top of security-only releases, starting with Magento 2.3.2-p2 and later. This issue was introduced by the new versioning scheme adopted for [security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches) starting with the {{ site.data.var.ee }} version 2.3.2-p2.
```

>    -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to break when applying patch sets on top of security-only releases, starting with Magento 2.3.2-p2 and later. This issue was introduced by the new versioning scheme adopted for [security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches) starting with the {{ site.data.var.ee }} version 2.3.2-p2.

Author: ```suggestion
   -  {:.fix}<!--MAGECLOUD-4661-->Fixed an issue that caused the `{{ site.data.var.ct }}` patching process to break when applying patch sets on top of security-only releases, starting with Magento version 2.3.2-p2 and later. This issue was introduced by the new versioning scheme adopted for [security-only patches]({{ site.baseurl }}/guides/v2.3/release-notes/bk-release-notes.html#security-only-patches) starting with the Magento version 2.3.2-p2.
```
